### PR TITLE
CNXML validation

### PR DESCRIPTION
**Pull request by pumazi**

### CODE REVIEW

File: cnxml/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# -*- coding: utf-8 -*-
+from .validation import *
```
Comment (reedstrm) [inline cnxml/__init__.py]: flake8 really doesn't like these, but I seem them in __init__ a lot, as way to hide details of implementation of an API. Can we either `# noqa` these, or I suppose I can try tweaking a local .flake8.cfg to ignore them.

File: cnxml/jing.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    'jing',
+)
+
+
+JING_JAR = lookup_resource('jing.jar')
+
+# This will map fatal messages to something human readable.
+KNOWN_FATAL_MESSAGES_MAPPING = {
+    'exception "java.io.IOException" thrown: Stream closed.':
+    'DOCTYPE declaration not allowed',
```
Comment (reedstrm) [inline cnxml/jing.py]: Yikes, this is really the error it throws for DOCTYPE? Perhaps stale-ware is right.

File: cnxml/__init__.py (hunk 1 above)
Comment (pumazi) [inline cnxml/__init__.py]: Hmm... I don't agree with flake8's analysis here. It's understood, or so I thought, that it's perfectly valid to `import *` in a package's `__init__.py` if the module has defined `__all__`.

File: cnxml/jing.py (hunk 2 above)
Comment (pumazi) [inline cnxml/jing.py]: This is cribbed from Rhaptos.CNXMLDocument's XMLService module.

**Comment by pumazi** [comment]
As far as a general name. The name is fairly general. At least, I've always referred to cnxml as the XML formats in Connexions, which is CNXML and COLLXML. But I suppose that would be an incorrect usage.

However, for the purposes of this exercise I don't know that the name matters too much, because nobody is going to see the name other than ourselves. Also, this is a throw away package. The XML validations we develop for HTML shouldn't be included here, because we don't want to untangle these bits later. Additionally, we can hopefully look at validation utilties that run on something up-to-date (because jing was built against jre-6 and jre-8 is the current stable).

We've headed down the road of calling it cnxml, registering it on pypi, documenting it in the planning docs and coding it with the name cnxml. I'd really hate to change all that stuff now...

**Comment by reedstrm** [comment]
Fair enough.  Should I merge? The set of commits seems reasonable, unless you want to squash some (typo fix, etc.)

**Pull request merged**